src/custom/pages/Swap/components/SwapButton/styled.tsx

```diff
@@ -0,0 +1,115 @@
+import styled from 'styled-components/macro'
+import { Dots } from 'components/swap/styleds'
+import { useCallback, useEffect, useState } from 'react'
+import { LONG_LOAD_THRESHOLD } from 'constants/index'
+import { ButtonSize, ThemedText } from 'theme'
+import { Text } from 'rebass'
+import { ButtonError, ButtonPrimary } from 'components/Button'
+import { Token } from '@uniswap/sdk-core'
+import { useReplaceSwapState, useSwapState } from 'state/swap/hooks'
+import { RowBetween } from '@src/components/Row'
+
+const fadeIn = `
+  @keyframes fadeIn {
+    0% {
+      opacity: 0;
+    }
+    100% {
+      opacity: 1;
+    }
+  }
+`
+
+const CenteredDots = styled(Dots)<{ smaller?: boolean }>`
+  vertical-align: ${({ smaller = false }) => (smaller ? 'normal' : 'super')};
+`
+
+const LongLoadText = styled.span`
+  animation: fadeIn 0.42s ease-in;
+
+  ${fadeIn}
+`
+
+type TradeLoadingProps = {
+  showButton?: boolean
+}
+
+export const TradeLoading = ({ showButton = false }: TradeLoadingProps) => {
+  const [isLongLoad, setIsLongLoad] = useState<boolean>(false)
+
+  // change message if user waiting too long
+  useEffect(() => {
+    const timeout = setTimeout(() => setIsLongLoad(true), LONG_LOAD_THRESHOLD)
+
+    return () => clearTimeout(timeout)
+  }, [])
+
+  const InsideContent = useCallback(
+    () => (
+      <ThemedText.Main display="flex" alignItems="center" maxHeight={20}>
+        <Text fontSize={isLongLoad ? 14 : 40} fontWeight={500}>
+          {isLongLoad && <LongLoadText>Hang in there. Calculating best price </LongLoadText>}
+          <CenteredDots smaller={isLongLoad} />
+        </Text>
+      </ThemedText.Main>
+    ),
+    [isLongLoad]
```

Comment:
Why is this inner component a callback?

Reply:
Right, got it now that's not your code, just moving it around.
Didn't look your style from what I've seen so far :)